app/views/upload.scala.html

```diff
@@ -1,6 +1,47 @@
 @()(implicit messages: Messages)
 
 @main(Messages("upload.header")) {
-    <h1 class="govuk-heading-xl">@Messages("upload.title")</h1>
-    <p>Placeholder page for upload</p>
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+        <span>
+            <a href="/dashboard" class="govuk-back-link">Back</a>
+            <span class="govuk-caption-xl">Step 3 of 3</span>
+            <h1 class="govuk-heading-xl">@Messages("upload.title")</h1>
+        </span>
+        <div class="govuk-form-group">
+            <label class="govuk-label" for="file-upload-1">
+              Select a folder
+            </label>
+            <input class="govuk-file-upload" id="file-upload-1" name="file-upload-1" type="file">
```

Comment:
Is the `-1` suffix necessary? You could call it "file-selection" or something, if you want to differentiate it from the rest of the page, since it's all about file upload?